client/src/components/tag-element-button/tag-element-button.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import classNames from 'classnames/bind'
+import * as DomTagging from '../../dom-tagging'
+// Styles
+import styles from './tag-element-button.scss'
+
+const css = classNames.bind(styles)
+
+class TagElementButton extends React.Component {
+  /*
+  constructor(props) {
+    super(props)
+    // TODO: Breaks react event handlers
+    DomTagging.hijackEventListeners()
+  }
+
+  componentDidMount() {
+    // TODO: Works but unnecessary if events are not logged (hijacked)
+    DomTagging.startObservingDomChange()
+  }
+  */
+
+  componentWillMount() {
+    const { elementTagged, toggleTagElementState } = this.props
+
+    DomTagging.setElementTaggedCallback(event => elementTagged(event))
+    DomTagging.setToggleTagElementStateCallback(() => toggleTagElementState())
```

Comment:
should these be somehow unbound?

Reply:
What do you mean?

Reply:
when the parent component is unmounted, `this.props.elementTagged` can become undefined. if the callback is called after the unmount, an error will occur.

Reply:
thanks!